test/e2e/operator.go

```diff
@@ -379,36 +387,125 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
+			Expect(err).NotTo(HaveOccurred())
 
-		err := wait.PollImmediate(30*time.Second, 10*time.Minute, muoIsDeployed)
-		Expect(err).NotTo(HaveOccurred())
+			By("getting the MUO deployment to delete")
+			deploymentToDelete := getMUODeployment(ctx)
+			Expect(deploymentToDelete).NotTo(BeNil())
+
+			By("deleting the MUO deployment")
+			err = deleteMUODeployment(ctx, deploymentToDelete.Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting for the MUO deployment to be reconciled")
+			err = wait.PollImmediate(interval, duration, doesMUODeploymentExistFn)
+			Expect(err).NotTo(HaveOccurred())
+		})
 	})
 })
 
+func getMUOPods(ctx context.Context) (*corev1.PodList, error) {
+	listOptions := metav1.ListOptions{
+		LabelSelector: "name=managed-upgrade-operator",
+	}
+	pods, err := clients.Kubernetes.CoreV1().Pods(managedUpgradeOperatorNamespace).List(ctx, listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pods.Items) != 1 {
+		return nil, fmt.Errorf("%d managed-upgrade-operator pods found", len(pods.Items))
+	}
+	return pods, nil
+}
+
+// getMUODeployment returns a pointer to the Managed Upgrade Operator
+// deployment if it exists. Otherwise it returns nil.
+// It does not return any error in case deployment is missing because this function
+// is aimed to be called multiple times while waiting for the MUO deployment to be created.
+// Returning an error would cause the polling/waiting mechanism to finish.
+func getMUODeployment(ctx context.Context) *appsv1.Deployment {
+	deployment, err := clients.Kubernetes.
+		AppsV1().
+		Deployments(managedUpgradeOperatorNamespace).
+		Get(ctx, managedUpgradeOperatorDeployment, metav1.GetOptions{})
+
+	if err != nil {
+		return nil
+	}
+
+	return deployment
+}
+
+func deleteMUODeployment(ctx context.Context, deploymentName string) error {
+	return clients.Kubernetes.
+		AppsV1().
+		Deployments(managedUpgradeOperatorNamespace).
+		Delete(ctx, deploymentName, metav1.DeleteOptions{})
+}
+
+func hasFIPSCryptoMandated(MUOLogs string) bool {
+	targetStr := `msg="FIPS crypto mandated: true"`
+	return strings.Contains(MUOLogs, targetStr)
+}
+
+func hasMUOFIPSCryptoMandatedHigherOrderFn(ctx context.Context) wait.ConditionFunc {
+	hasMUOFIPSCryptoMandatedFn := func() (bool, error) {
+		By("getting MUO pods")
+		MUOPods, err := getMUOPods(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		By("getting logs from MUO")
+		MUOPodName := MUOPods.Items[0].Name
+		rawLogs, err := clients.Kubernetes.
+			CoreV1().
+			Pods(managedUpgradeOperatorNamespace).
+			GetLogs(MUOPodName, &corev1.PodLogOptions{}).
+			DoRaw(ctx)
+
+		if err != nil {
+			return false, err
+		}
+
+		By("verifying that MUO has FIPS crypto mandated by reading logs")
+		logs := string(rawLogs)
+		return hasFIPSCryptoMandated(logs), nil
+	}
+
+	return hasMUOFIPSCryptoMandatedFn
+}
+
+// doesMUODeploymentExistHigherOrderFn returns a function which does not return any error as it
```

Comment:
couple typos in there: mechanisnm, does not exists, comma missing before the first and

Reply:
Done. Tks!